OrbitCaptureClient/CaptureEventProcessor.cpp

```diff
@@ -211,6 +213,33 @@ uint64_t CaptureEventProcessor::GetCallstackHashAndSendToListenerIfNecessary(
   return hash;
 }
 
+void CaptureEventProcessor::ProcessInternedTracepointInfo(
+    orbit_grpc_protos::InternedTracepointInfo interned_tracepoint_info) {
+  if (tracepoint_intern_pool_.contains(interned_tracepoint_info.key())) {
+    ERROR("Overwriting InternedTracepointInfo with key %llu", interned_tracepoint_info.key());
+  }
+  tracepoint_intern_pool_.emplace(interned_tracepoint_info.key(),
+                                  std::move(*interned_tracepoint_info.mutable_intern()));
+}
+void CaptureEventProcessor::ProcessTracepointEvent(
+    const orbit_grpc_protos::TracepointEvent& tracepoint_event) {
+  orbit_grpc_protos::TracepointInfo tracepoint_info;
+
+  CHECK(tracepoint_event.tracepoint_info_or_key_case() ==
+        orbit_grpc_protos::TracepointEvent::kTracepointInfoKey);
+  tracepoint_info = tracepoint_intern_pool_[tracepoint_event.tracepoint_info_key()];
+
+  uint64_t hash = GetTracepointInfoHashAndSendToListenerIfNecessary(tracepoint_info);
```

Comment:
I don't see why we need to "re-hash": Can't we just use `tracepoint_event_info.set_tracepoint_info_key(tracepoint_event.tracepoint_info_key());`
and propagate the `tracepoint_intern_pool_` to whoever is interested?

Reply:
I have noticed that the hash mechanism is different in the client-side (even the names of the thread tracks are hashed differently , which are just strings) and I wanted to keep it consistent. I will have a chat with someone to make sure that the different hash mechanism is justified. However, of course, I can just use 'tracepoint_event.tracepoint_info_key()' as the hash for the tracepoint event.

Reply:
There are a couple of reasons why also the other methods here re-hash but in general I guess it's mostly simplicity. This allows to have the intern pools local to the `CaptureEventProcessor` that handle interning during the communication, then other pools (not called like that anymore, e.g., `StringManager`) filled by the `CaptureListener` closer to where the strings are used by the UI (e.g., the capture view). Also keep in mind that the `CaptureEventProcessor` completely translates the protos for the communication (capture.proto) to the protos for the client (capture_data.proto). I think we discussed this problem a few weeks ago and didn't really come to a conclusion.

Reply:
@dpallotti thanks for explaining! In this case, do you think it is justified to re-hash for keeping the intern pool local to the CaptureEventProcessor? I am also planning of having another pool for registering the tracepoints events (such as TracepointEventInfoManager) in another CL.

Reply:
Since for callstacks and strings we already re-hash and duplicate the intern pool mechanism, for me it's fine to do the same for `TracepointInfo`s. If you want, you could at least avoid re-computing the hash, by changing `GetTracepointInfoHashAndSendToListenerIfNecessary` to `SendTracepointInfoToListenerIfNecessary` and passing a second argument `hash` with the hash that you already have (`tracepoint_event.tracepoint_info_key()`).

Reply:
@danielfenner what do you think?

Reply:
For now, I have done what @dpallotti suggested to avoid re-hashing.

Reply:
Let's stick with "re-hashing" for now. Since all of this happens in the UI I am not too concerned about performance. If that hits us we can refactor later.
Plus it might be a bit weird to have the hash map from the service duplicated here since we can't reproduce the hash (we use std::hash in the service and hash functions are only required to produce the same result for the same input within a single execution of a program).

Reply:
Of course! Thank you @danielfenner!

Reply:
As discussed offline, we will keep without the re-hash.